pinot-core/src/main/java/org/apache/pinot/core/indexsegment/immutable/ImmutableSegmentImpl.java

```diff
@@ -60,7 +60,11 @@ public ImmutableSegmentImpl(SegmentDirectory segmentDirectory, SegmentMetadataIm
 
   @Override
   public Dictionary getDictionary(String column) {
-    return _indexContainerMap.get(column).getDictionary();
+    ColumnIndexContainer container = _indexContainerMap.get(column);
+    if (container == null) {
```

Comment:
```
Preconditions.checkNotNull(container, "Invalid column: " + column);
```

Reply:
@snleee, was in pending state for too many days. Better to get the functionality committed asap